src/main/java/io/vertx/core/http/RequestOptions.java

```diff
@@ -333,7 +333,19 @@ private URL parseUrl(String surl) {
   public RequestOptions setAbsoluteURI(String absoluteURI) {
     Objects.requireNonNull(absoluteURI, "Cannot set a null absolute URI");
     URL url = parseUrl(absoluteURI);
-    Boolean ssl = false;
+    return setAbsoluteURI(url);
+  }
+
+  /**
+   * Like {@link #setAbsoluteURI(String)} but using an {@link URL} parameter.
+   *
+   * @param url the uri to use
+   * @return a reference to this, so the API can be used fluently
+   */
+  @GenIgnore
+  public RequestOptions setAbsoluteURI(URL url) {
```

Comment:
`URL` used to give trouble with native images compiled without ssl support. if this is switched to `URI` it is always known to work and it matches better with the method name.

Reply:
Internally, we use `URL`. The original issue was filed by a user who works with `URL`. If users have a `URI` they can still make the conversion.

Do you know if there is an issue tracking the problem you mentioned? In practice, would `native-image` with just any program that has a method with `URL` parameter? Or only programs that actually use that method?

In the latter case, I believe we can tell users to avoid using that method as a workaround.